base_decorator.rb

```diff
@@ -0,0 +1,10 @@
+class BaseDecorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
```

Comment:
[OPTIONAL] I think it would be best if you also added the `attr_accessor` for the nameable variable so that you have a getter and setter for it.